lib/core_extensions.rb

```diff
@@ -39,8 +39,14 @@ def initialize(*attribute)
 
     def before_destroy(record)
       klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
+        record.association(klass.to_sym).association_scope.each do |what|
```

Comment:
I'd go for okay on this, and if it doesn't work on rails4 we'll adjust on that branch. I think unscoped should work on 4 as well as on 3, but not 100% sure. it's covered in a test, so we'll see if it breaks.